core/api/src/main/java/org/opennms/core/config/api/TextEncryptor.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2020 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2020 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.core.config.api;
+
+/**
+ * For a given context/alias, encrypt/decrypt text with specified encryptionKey.
```

Comment:
I find the use of the `encryptionKey` here a little confusing.

We hash the key with some salt using `StrongPasswordEncryptor` to generate a unique password - this password is stored in SCV and we subsequently use that to encrypt/decrypt the payload.

If we call `decrypt` after `encrypt` with a given key, the key is ignored (since the actual password is retrieved from SCV).
If we call `encrypt` with a different key, but the same alias, then original key is still used.

Given this, it seems like the key can be dropped completed and we can just generate a random password for a given alias and store/retrieve this in SCV.

Reply:
Yeah, initially I was saving the key,value in `Credentials` attributes. Since that's moved to Credentials and there is no user input for the key guess that can be dropped.